source/common/stream_info/stream_info_impl.h

```diff
@@ -253,13 +253,17 @@ struct StreamInfoImpl : public StreamInfo {
 
   const Http::RequestHeaderMap* getRequestHeaders() const override { return request_headers_; }
 
-  void setRequestIDExtension(Http::RequestIDExtensionSharedPtr utils) override {
-    request_id_extension_ = utils;
+  void setRequestIDProvider(const Http::RequestIdStreamInfoProviderSharedPtr& provider) override {
+    ASSERT(provider != nullptr);
```

Comment:
Is this `ASSERT` or `ENVOY_BUG`?

Reply:
ENVOY_BUG seems unnecessary to me for this one but will switch if you prefer.